lib/hubspot/company.rb

```diff
@@ -39,18 +39,35 @@ def all(opts={})
       end
 
       # Finds a list of companies by domain
-      # {http://developers.hubspot.com/docs/methods/companies/get_companies_by_domain}
+      # {https://developers.hubspot.com/docs/methods/companies/search_companies_by_domain}
       # @param domain [String] company domain to search by
+      # @param options [Hash] Possible options are:
+      #    limit [Integer] for pagination
+      #    properties [Array] list of company properties to recieve
+      #    offset_company_id [Integer] for pagination (should be company ID)
       # @return [Array] Array of Hubspot::Company records
-      def find_by_domain(domain)
-        path = GET_COMPANY_BY_DOMAIN_PATH
-        params = { domain: domain }
-        raise Hubspot::InvalidParams, 'expecting Integer parameter' unless domain.try(:is_a?, String)
+      def find_by_domain(domain, options = {})
+        raise Hubspot::InvalidParams, 'expecting String parameter' unless domain.try(:is_a?, String)
+
+        limit = options.fetch(:limit, 100)
+        properties = options.fetch(:properties) { Hubspot::CompanyProperties.all.map { |property| property["name"] } }
```

Comment:
When I compared this against the `master` branch using one of our contacts, I found that this generates almost the same response. The difference is that the old endpoint gave two properties that weren't present in this new lookup: `hs_analytics_first_touch_converting_campaign` and `hs_analytics_last_touch_converting_campaign`. I'm not sure why those two properties aren't included in the "all" properties but when I added them to the properties collection, I received the same response as the current implementation. Since this version doesn't include support for campaigns, I think we can safely skip these properties.